src/order/irreducible.lean

```diff
@@ -0,0 +1,189 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.lattice
+import data.fintype.card
+
+/-!
+# Birkhoff's representation theorem
+-/
+
+section
+variables {α : Type*} [preorder α] [finite α]
+
+@[priority 100] -- See note [lower instance priority]
```

Comment:
Shouldn't these go somewhere else, like next to `finite.preorder.well_founded_lt`